src/main/java/org/hillel/homework/app/ShelterApp.java

```diff
@@ -0,0 +1,183 @@
+package org.hillel.homework.app;
+
+import org.hillel.homework.exception.InvalidJSONException;
+import org.hillel.homework.service.ShelterService;
+import org.hillel.homework.utils.Validator;
+import org.hillel.homework.model.Pet;
+import org.hillel.homework.utils.constants.Constants;
+import java.io.IOException;
+import java.util.List;
+import java.util.Scanner;
+
+public class ShelterApp {
+    private final Scanner scanner = new Scanner(System.in);
+    private final ShelterService service;
+
+    public ShelterApp(ShelterService service) throws IOException {
+        this.service = service;
+    }
+
+    public int menu() {
+        display(Constants.MENU);
+
+        int res;
+        while (true) {
+            display(Constants.MENU_OPERATION_PROMPT, false);
+            String strRes = input();
+            if (Validator.stringNotEmpty(strRes)) {
+                if (Validator.stringNotSpaces(strRes)) {
+                    if (Validator.isWholeNumber(strRes)) {
+                        if (Validator.wholeNumberRange("1", "4", strRes)) {
+                            res = Integer.parseInt(strRes);
```

Comment:
if if if if is are not a good code style. maybe it will be better to move all this logic to the private method